### ISSUE TITLE
TOTP Compatibility with KeeWeb and KeePassXC

### ISSUE THREAD

**Issue by dblitt99** [strongbox-password-safe/Strongbox#233]
I am trying to use TOTP in my database and I am unable to see the generated TOTP codes in Strongbox when they are not inserted by Strongbox. It seems that Strongbox only supports the format of
```
TOTP Seed: ZOFHRYXNSJDUGHSJ
TOTP Settings: 30;6
```
However, KeeWeb creates TOTP entries with the following formats:
```
otp: key=ZOFHRYXNSJDUGHSJ
```
```
otp: otpauth://totp/default?secret=ZOFHRYXNSJDUGHSJ
```

Both of these formats work with KeeWeb and KeePassXC, but they are not recognized by Strongbox.

**Comment by dblitt99** [comment]
It appears that the above formats were fixed in the last version of the iOS app and I was not up to date. I believe that it is still not fixed in the macOS app. However, it appears that Strongbox still does not support the following TOTP formats:
```
otp: ZOFHRYXNSJDUGHSJ
```
```
otp: ZOFH RYXN SJDU GHSJ
```
```
otp: zofhryxnsjdughsj
```
```
otp: zofh ryxn sjdu ghsj
```

**Comment by mmcguill** [comment]
Hi @dblitt99 - Thanks for the report. Yes 1.44.4 should support the new format from KeePassXC. 

I haven't seen those other 4 formats you posted before, is there a list of the formats you know about somewhere?

Re: Mac version - Yes, aware of that, will release a new build shortly to fix

**Comment by dblitt99** [comment]
No, I don't have a list - those are just some formats I've encountered. I think it happened when I just pasted the value into the otp field and KeeWeb went with it.

**Comment by mmcguill** [comment]
Ok, I'd prefer not to add support for all these different schemes, there are already 3 or 4 different ways of adding TOTP settings... Ideally we'd all try to sync on the new KeePassXC style which allows for full customization of the TOTP settings...

**Comment by dblitt99** [comment]
While I agree that ideally everything would be better if it all followed a single style, I (and possibly others) would really benefit from Strongbox supporting the same schemes that KeeWeb and KeePassXC support.

**Comment by mmcguill** [comment]
So, I've been playing around with these formats (quite curious where you came across them actually?) - while KeePassXC does look to accept them, I'm fairly sure the generated code is not quite correct...

You can check this by creating a record with:

**Standard KeeOTP Plugin format**
otp:key=ZOFHRYXNSJDUGHSJ

**and your example(s) from above**
otp:ZOFHRYXNSJDUGHSJ

In KeePassXC 2.5.1 these generate different codes for me... 

I believe what's happening here is that your format is not actually supported but a code is generated anyway based on a "default" secret, possibly null/empty.

I've added support for your formats to Strongbox in the meantime, but I hated life while doing it...

**Comment by dblitt99** [comment]
Hmm - my testing is showing that you're correct - I'm really not sure what KeePassXC is seeing but it definitely isn't the correct TOTP code. KeeWeb for sure gives me the correct code.

**Comment by mmcguill** [comment]
Ok, might be a good idea to open this bug over with the KeePassXC guys, I'm sure they'll want to support these weird formats too... This is available now with 1.44.5 (iOS) and 1.11.3 (Mac)... going to close this one out, feel free to re-open if you have any issues.

**Comment by dblitt99** [comment]
That's probably a good idea. Thanks for taking care of this!
